src/unpack.rs

```diff
@@ -163,13 +163,14 @@ async fn create_prefix(channel_dir: &Path, target_prefix: &Path) -> Result<()> {
         .await
         .map_err(|e| anyhow!("could not collect packages: {}", e))?;
 
-    let cache_dir = tempfile::tempdir()
-        .map_err(|e| anyhow!("could not create temporary directory: {}", e))?
-        .into_path();
+    let tmp_dir =
+        tempfile::tempdir().map_err(|e| anyhow!("Could not create temporary directory: {}", e))?;
+    let cache_dir = tmp_dir.path();
```

Comment:
Oh, I [see](https://docs.rs/tempfile/latest/tempfile/struct.TempDir.html#method.into_path):

> `into_path` [...] consumes the `TempDir` without deleting directory on the filesystem, meaning that the directory will no longer be automatically deleted.

Good catch, thanks!

We should also be aware of https://docs.rs/tempfile/latest/tempfile/struct.TempDir.html#resource-leaking:

> It’s important to ensure that handles (like `File` and `ReadDir`) to files inside the directory are dropped before the `TempDir` goes out of scope. The `TempDir` destructor will silently ignore any errors in deleting the directory; to instead handle errors call `TempDir::close()`.

I assume the handles will go out of scope together with the `TempDir` and will be destroyed first? Did you test that?

Should we explicitly `::close()` the `TempDir` and check for errors?